scipy/optimize/_lsap_module.c

```diff
@@ -28,21 +28,24 @@ THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */
 
+#include <math.h>
 #include "numpy/arrayobject.h"
 #include "numpy/ndarraytypes.h"
 #include "rectangular_lsap/rectangular_lsap.h"
 
 static PyObject*
 calculate_assignment(PyObject* self, PyObject* args)
 {
-    PyObject* x = NULL;
+    PyObject* a = NULL;
+    PyObject* b = NULL;
+    PyObject* result = NULL;
     PyObject* obj_cost = NULL;
     if (!PyArg_ParseTuple(args, "O", &obj_cost))
         return NULL;
 
     PyArrayObject* obj_cont =
       (PyArrayObject*)PyArray_ContiguousFromAny(obj_cost, NPY_DOUBLE, 2, 2);
-    if (obj_cont == NULL) {
+    if (!obj_cont) {
         PyErr_SetString(PyExc_TypeError, "invalid cost matrix object");
```

Comment:
@peterbell10 should I also propagate the exception here or is this OK?

Reply:
I can see arguments for both ways, so leaving it as is is probably fine.